lucet-runtime/lucet-runtime-internals/src/instance/signals.rs

```diff
@@ -255,11 +258,7 @@ extern "C" fn handle_signal(signum: c_int, siginfo_ptr: *mut siginfo_t, ucontext
     });
 
     if switch_to_host {
-        HOST_CTX.with(|host_ctx| unsafe {
-            Context::set_from_signal(&*host_ctx.get())
-                .expect("can successfully switch back to the host context");
-        });
-        unreachable!()
+        ctx.set_ip(Context::set_from_signal as *const c_void);
```

Comment:
Could you add a little prose comment for our future selves about what's being done here, and why we're not calling `set_from_signal` directly?